Backend/Backend/Services/BaseDataService.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.EntityFrameworkCore;
+
+namespace Backend.Services
+{
+    public abstract class BaseDataService<T>
+        where T : DbContext
+    {
+        private readonly T _dbContext;
+        private readonly ILogger<T> _logger;
+
+        public BaseDataService(
+            T dbContext,
+            ILogger<T> logger)
+        {
+            _dbContext = dbContext;
+            _logger = logger;
+        }
+
+        protected async Task<T1> ExecuteSafe<T1>(Func<Task<T1>> action)
```

Comment:
For naming of generic type parameters, its bad practise to name T1,T2. 
In MsDocs https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces?redirectedfrom=MSDN
advise to use Prefix naming